README.md

```diff
@@ -1,2 +1,145 @@
+            ]
+        ),
+        .target(name: "Run", dependencies: [
+            .target(name: "App"),
+            .product(name: "MongoDBVapor", package: "mongodb-vapor")
+        ])
+    ]
+)
+```
+
+Then run `swift build` to download, compile, and link all your dependencies.
+
+## Example Usage
+### You can find a complete example project built using this library [here](https://github.com/mongodb/mongo-swift-driver/tree/main/Examples/VaporExample). 
+
+To summarize the available features:
+
+### Configure global settings
+
+In `Run/main.swift`, add:
+```swift
+import MongoDBVapor
+
+// Configure the app for using a MongoDB server at the provided connection string.
+try app.mongoDB.configure("mongodb://localhost:27017")
+
+defer {
+    // Cleanup the application's MongoDB data.
+    app.mongoDB.cleanup()
+    // Clean up the driver's global state. The driver will no longer be usable from this program after this method is
+    // called.
+    cleanupMongoSwift()
+}
+```
+
+### Use MongoDB in a Request Handler
+For collections you plan to access frequently, we recommend adding computed properties in an extension to `Request`
+to provide easy access, like:
+```swift
+extension Request {
+    /// A collection with an associated `Codable` type `Kitten`.
+    var kittenCollection: MongoCollection<Kitten> {
+        self.client.db("home").collection("kittens", withType: Kitten.self)
+    }
+}
+```
+
+**Any client, database, or collection object you access via `Request.mongoDB` will automatically return `EventLoopFuture`s 
+on the same `EventLoop` the `Request` is being handled on, simplifying thread safety concerns and improving performance
+by removing the need to `hop` the returned futures.**
+
+You can then use these in request handlers as follows:
+```swift
+/// Handles a request to load the list of kittens.
+app.get("kittens") { req -> EventLoopFuture<[Kitten]> in
+    req.kittenCollection.find().flatMap { cursor in
+        cursor.toArray()
+    }
+}
+
+app.post("kittens") { req -> EventLoopFuture<Response> in
+    let newKitten = try req.content.decode(Kitten.self)
+    return req.kittenCollection.insertOne(newKitten)
+        .map { _ in Response(status: .created) }
+}
+```
+
+### Perform one-time setup or teardown code
+If you have one-time code you'd like to run each time your application starts up, e.g. in `Run/main.swift`, you can
+use the global client, accessible via `Application.mongoDB`:
+```swift
+// Configure the app for using a MongoDB server at the provided connection string.
+try app.mongoDB.configure("mongodb://localhost:27017")
+let coll = app.mongoDB.client.db("home").collection("kittens")
+// creates a unique index if it doesn't exist already.
+_ = try coll.createIndex(["name": 1], indexOptions: IndexOptions(unique: true)).wait()
+```
+
+### Working with Extended JSON
+If you'd like to use `ExtendedJSONEncoder` and `ExtendedJSONDecoder` for encoding/decoding JSON requests/responses, in  `App/configure.swift`, add:
```

Comment:
I think it's probably worth pointing out that if you want to use a `BSONDocument` directly as `Content`, you'll need to do this configuration. Once we implement regular `Codable` conformance we can remove the warning.

Reply:
added; making follow-up PR to BSON repo to update JSON guide.

Reply:
actually looks like we already have all of that there - Nellie had added a section which I updated a couple weeks ago that has basically the same content and notes not all BSON types work with JSONEncoder/JSONDecoder

https://github.com/mongodb/swift-bson/blob/main/Guides/JSON-Interop.md#using-extended-json-with-vapor